src/modules/keyboardmanager/KeyboardManagerEditorLibrary/BufferValidationHelpers.cpp

```diff
@@ -23,9 +24,16 @@ namespace BufferValidationHelpers
             // Check if the value being set is the same as the other column
             if (remapBuffer[rowIndex].first[std::abs(int(colIndex) - 1)].index() == 0)
             {
-                if (std::get<DWORD>(remapBuffer[rowIndex].first[std::abs(int(colIndex) - 1)]) == selectedKeyCode)
+                DWORD otherColumnKeyCode = std::get<DWORD>(remapBuffer[rowIndex].first[std::abs(int(colIndex) - 1)]);
+                if (otherColumnKeyCode == selectedKeyCode)
                 {
                     errorType = ShortcutErrorType::MapToSameKey;
+                } else if (
+                    (otherColumnKeyCode == Helpers::GetCombinedKey(selectedKeyCode) || selectedKeyCode == Helpers::GetCombinedKey(selectedKeyCode)) &&
+                    Helpers::GetCombinedKey(otherColumnKeyCode) == Helpers::GetCombinedKey(selectedKeyCode)
```

Comment:
As I understand we want to handle the following cases here:
`X (Something) -> X (Something)`. Where `X` is the modifier key and `Something` is `Left`, `Right` or empty. Is it correct?

For me more natural check would be `Helpers::IsModifierKey(selectedKeyCode) && Helpers::GetCombinedKey(otherColumnKeyCode) == Helpers::GetCombinedKey(selectedKeyCode)`.

Also, it's a good idea to move this check into a separate method with descriptive name

Reply:
We want to handle the following cases here
```
X (Left) -> X
X -> X (Left)
X (Right) -> X
X -> X (Right)
```

Is it correct?

Reply:
In this case we can replace `otherColumnKeyCode == Helpers::GetCombinedKey(selectedKeyCode)` to `otherColumnKeyCode == Helpers::GetCombinedKey(otherColumnKeyCode )`?

Reply:
> X (Left) -> X
> X -> X (Left)
> X (Right) -> X
> X -> X (Right)

This is correct.

Reply:
I've pushed the requested changes.